api/src/database/postgis/models/layerModel.ts

```diff
@@ -30,8 +30,8 @@ const getMapLayersWithGeoData = async (db = getDb()): Promise<Array<MapLayerWith
     .innerJoin(LAYER_GEO_DATA_TABLE, function () {
       this.on(`${MAP_LAYERS_TABLE}.geo_reference_id`, '=', `${LAYER_GEO_DATA_TABLE}.reference_id`);
     })
-    .innerJoin(ATTRIBUTES_TABLE, function () {
-      this.on(`${MAP_LAYERS_TABLE}.geo_reference_id`, '=', db.raw(`${ATTRIBUTES_TABLE}.geo_data->>'referenceId'`));
+    .leftJoin(ATTRIBUTES_TABLE, function () {
```

Comment:
Hi, what is a difference between using:
```
    . leftJoin(ATTRIBUTES_TABLE, function () {
      this.on(`${MAP_LAYERS_TABLE}.geo_reference_id`, '=', db.raw(`${ATTRIBUTES_TABLE}.geo_data->>'referenceId'`));
```
      
and
      
```
      .leftJoin(ATTRIBUTES_TABLE, function () {
this.on(`${MAP_LAYERS_TABLE}.attribute_id`, '=', `${ATTRIBUTES_TABLE}.attribute_id`);})
```

here? The first will fail because there is no record in geo_data column for tile layers so I cannot access referenceId of it? By the second option I should get all rows from mapLayers table(with left join). This function was originally meant to filter mapLayers only for the ones with geoData in file or in database. So which layers shouldn't be returned from this function now?

Reply:
The first has more issues, as the geo_data.geoReferenceId is more like informational value, missing in most of data we are currently working with

the new query uses primary key on the table. The left join to include also layers with no attributeId (and often reason for it) like tile layers

Reply:
thanks for explanation